src/metabase/cmd/env_var_dox.clj

```diff
@@ -0,0 +1,110 @@
+  []
+  (for [ns-symb (ns.find/find-namespaces (classpath/system-classpath))
+        :when (and
+               (str/includes? (name ns-symb) "metabase")
+               (not (str/includes? (name ns-symb) "test")))]
+    (require ns-symb))
+  (seq (into (sorted-map) @setting/registered-settings)))
+
+;;;; Filter functions
+
+(defn- avoid?
+  "Used to filter out environment variables with high foot-gun indices."
+  [env-var]
+  (contains? env-vars-not-to-mess-with (u/screaming-snake-case (name (:name env-var)))))
+
+(defn- setter?
+  "Used to filter out environment variables that cannot be set."
+  [env-var]
+  (not= :none (:setter env-var)))
+
+(defn- active?
+  "Used to filter our deprecated enviroment variables."
+  [env-var]
+  (nil? (:deprecated env-var)))
+
+;;;; Formatting functions
+
+(defn- format-type
+  "Helper function to specify the format of an environment variable's type for its documentation."
+  [env-var]
+  (str "Type: " (name (:type env-var))))
+
+(defn- format-default
+  "Helper function to specify how to format the default value of an enviromnent variable.
+  for use in the environment variable docs."
+  [env-var]
+  (let [d (:default env-var)]
+    (str "Default: "
+         (if (false? d) "`false`"
+             (if (:default env-var)
+               (str "`" (:default env-var) "`")
+               "`null`")))))
+
+(defn- format-prefix
+  "Used to build an environment variable."
+  [s]
+  (str "MB_" s))
+
+(defn- format-heading
+  "Takes an integer and a string and creates a Markdown heading of level n."
+  [n s]
+  (str (apply str (take n (repeat "#"))) " `" s "`"))
+
+(defn- format-description
+  "Helper function to specify description format for enviromnent variable docs."
+  [env-var]
+  (->> (:description env-var)
+       u/add-period
+       ;; Drop brackets used to create source code links
+       (#(str/replace % #"\[\[|\]\]" ""))))
+
+(defn format-env-var-entry
+  "Preps a doc entry for an environment variable as a Markdown section."
+  [env-var]
+  (str/join "\n\n" [(format-heading 3 (format-prefix (u/screaming-snake-case (name (:name env-var)))))
+                    (format-type env-var)
+                    (format-default env-var)
+                    (format-description env-var)]))
+
+(defn format-env-var-docs
+  "Preps relevant environment variable docs as a Markdown string."
+  []
+  (->> (get-settings)
+       (map (fn [[_ v]] v))
+       (filter setter?)
+       (filter active?)
+       (remove avoid?)
+       (map format-env-var-entry)))
+
+(defn- format-intro
+  "Exists just so we can write the intro in Markdown."
+  []
+  (str (slurp "src/metabase/cmd/resources/env-var-intro.md") "\n\n"))
+
+(defn generate-dox!
+  "Prints the generated environment variable docs to a file."
+  []
+  (println "Generating docs for environment variables...")
+  (spit (io/file "docs/configuring-metabase/environment-variables.md") (apply str (format-intro)
```

Comment:
```suggestion
  (spit (io/file "docs/configuring-metabase/environment-variables.md") (str (format-intro)
```

`apply` takes the last argument and treats it as a collections, so for example 

```
(apply + [1 2 3])
=>
(+ 1 2 3)
=>
6
```

When you do that with a string it splits it out into separate characters (Clojure syntax is like `\a` for the character `a`), so it does something like

```clj
(apply str "Hello" "World")
=>
(str "Hello" \W \o \r \l \d)
=> 
"HelloWorld"
```

You don't need or want to do it that way. Because `str` will combine its arguments into a single string either way.

```clj
(str "Hello" "World")
=>
"HelloWorld"
```

works just fine